caddyconfig/caddyfile/dispenser.go

```diff
@@ -343,15 +343,28 @@ func (d *Dispenser) EOFErr() error {
 	return d.Errf("Unexpected EOF")
 }
 
+type parseError struct {
+	file       string
+	line       int
+	innerError error
+}
+
+func (e parseError) Error() string {
+	return fmt.Sprintf("%s:%d - Error during parsing: %s", e.file, e.line, e.innerError.Error())
+}
+
+func (e parseError) Unwrap() error {
+	return e.innerError
+}
+
 // Err generates a custom parse-time error with a message of msg.
 func (d *Dispenser) Err(msg string) error {
-	msg = fmt.Sprintf("%s:%d - Error during parsing: %s", d.File(), d.Line(), msg)
-	return errors.New(msg)
+	return parseError{d.File(), d.Line(), errors.New(msg)}
```

Comment:
If Err could use Errf, or vice-versa, that might be nice to reduce a little bit of code duplication / repetition.

Reply:
I refactored it and I think it's more concise now.